installer/x86_64/install.sh

```diff
@@ -381,49 +389,74 @@ demo_install_uefi_grub()
 
 }
 
-eval $create_demo_partition $blk_dev
-demo_dev=$(echo $blk_dev | sed -e 's/\(mmcblk[0-9]\)/\1p/')$demo_part
+if [ "$environment" != "sonic" ]; then
+    eval $create_demo_partition $blk_dev
+    demo_dev=$(echo $blk_dev | sed -e 's/\(mmcblk[0-9]\)/\1p/')$demo_part
 
-# Make filesystem
-mkfs.ext4 -L $demo_volume_revision_label $demo_dev
+    # Make filesystem
+    mkfs.ext4 -L $demo_volume_label $demo_dev
 
-# Mount demo filesystem
-demo_mnt=$(${onie_bin} mktemp -d) || {
-    echo "Error: Unable to create file system mount point"
-    exit 1
-}
-trap_push "${onie_bin} fuser -km $demo_mnt || ${onie_bin} umount $demo_mnt || ${onie_bin} rmdir $demo_mnt || true"
-${onie_bin} mount -t ext4 -o defaults,rw $demo_dev $demo_mnt || {
-    echo "Error: Unable to mount $demo_dev on $demo_mnt"
-    exit 1
-}
+    # Mount demo filesystem
+    demo_mnt=$(${onie_bin} mktemp -d) || {
+        echo "Error: Unable to create file system mount point"
+        exit 1
+    }
+    trap_push "${onie_bin} fuser -km $demo_mnt || ${onie_bin} umount $demo_mnt || ${onie_bin} rmdir $demo_mnt || true"
+    ${onie_bin} mount -t ext4 -o defaults,rw $demo_dev $demo_mnt || {
+        echo "Error: Unable to mount $demo_dev on $demo_mnt"
+        exit 1
+    }
+else
+    demo_mnt="/host"
+fi
+
+sonic_dir="$SONIC_IMAGE_DIR_PREFIX-$git_revision"
+
+echo "Installing SONiC to $demo_mnt/$sonic_dir"
+
+# Create target directory or clean it up if exists
+if [ -d $demo_mnt/$sonic_dir ]; then
+    echo "Directory $demo_mnt/$sonic_dir/ already exists. Cleaning up..."
+    rm -rf $demo_mnt/$sonic_dir/*
+else
+    mkdir $demo_mnt/$sonic_dir || {
+        echo "Error: Unable to create sonic directory"
+        exit 1
+    }
+fi
 
 # Decompress the file for the file system directly to the partition
-unzip $ONIE_INSTALLER_PAYLOAD -d $demo_mnt
+unzip $ONIE_INSTALLER_PAYLOAD -d $demo_mnt/$sonic_dir
 
-if [ -f $demo_mnt/$FILESYSTEM_DOCKERFS ]; then
-    cd $demo_mnt && mkdir -p $DOCKERFS_DIR && tar xf $FILESYSTEM_DOCKERFS -C $DOCKERFS_DIR; cd $OLDPWD
+if [ -f $demo_mnt/$sonic_dir/$FILESYSTEM_DOCKERFS ]; then
+    cd $demo_mnt/$sonic_dir && mkdir -p $DOCKERFS_DIR && tar xf $FILESYSTEM_DOCKERFS -C $DOCKERFS_DIR && rm $FILESYSTEM_DOCKERFS; cd $OLDPWD
 fi
 
 # Create loop device for /var/log to limit its size to $VAR_LOG_SIZE MB
-if [ "$VAR_LOG_SIZE" != "0" ]; then
-    mkdir -p $demo_mnt/disk-img
-    dd if=/dev/zero of=$demo_mnt/disk-img/var-log.ext4 count=$((2048*$VAR_LOG_SIZE))
-    mkfs.ext4 -q $demo_mnt/disk-img/var-log.ext4 -F
+if [ ! -f $demo_mnt/disk-img/var-log.ext4 ]; then
+    if [ "$VAR_LOG_SIZE" != "0" ]; then
+        mkdir -p $demo_mnt/disk-img
+        dd if=/dev/zero of=$demo_mnt/disk-img/var-log.ext4 count=$((2048*$VAR_LOG_SIZE))
+        mkfs.ext4 -q $demo_mnt/disk-img/var-log.ext4 -F
+    fi
+else
+    echo "Log file system already exists. Skipping..."
 fi
```

Comment:
can we also check the var-log size since the new sonic image may increase or descreae the file size. 

Can we compare the size, if they are not the same then try to create a new one with new size.